### ISSUE TITLE
Will the rowkey with compaction in cache be load after tsdb restart

### ISSUE THREAD

**Issue by micklonge** [OpenTSDB/opentsdb#911]
Hi, team.
     the rowkey in compactionqueue is generated by the PUT op。they will be cached.
     However，if the tsd restart，the rowkey in cache will be lost。
     Will they be load after restart?

**Comment by manolama** [comment]
No, they aren't reloaded after a TSD restart so those rows will not be compacted unless they're queried (TSDs can compact rows when read as well).

**Comment by c61811** [comment]
What's the fix this problem, please?Is how to solve? @manolama

**Comment by manolama** [comment]
Well, assuming the goal is to reduce storage overhead, there are three and 1/2 options here:
1) We could have an HBase co-proc that compacts the columns so the TSD's don't have to.
2) Switch to appends and that will obviate the need for compactions. It's just more expensive (see the 1/2 below).
3) Modify the fsck tool so that it will compact. I *think* it might already, we'd have to check. Then folks can scan the hours or so after the TSD tanked.
1/2) The Yahoo append co-proc if we can ever get that out will make appends much more friendly.

Alternatively we *could* write the row keys to a log but that's a little bit of a pain. @c61811 want to try implementing anything?

**Comment by c61811** [comment]
@manolama I want  to try implementing something very much.  I'm from JD.com in Big data development department. Currently we are having more usage scenarios with OpenTSDB.
Please contact to me!
